src/points/PointsHistoryBottomSheet.tsx

```diff
@@ -79,9 +109,16 @@ function PointsHistoryBottomSheet({ forwardedRef }: Props) {
     return groupFeedItemsInSections([], pointsHistory, (t: ClaimHistory) => Date.parse(t.createdAt))
   }, [pointsHistory])
 
-  // TODO: Render items
-  const renderItem = () => {
-    return <></>
+  const renderItem: ListRenderItem<ClaimHistory> = ({ item }: { item: ClaimHistory }) => {
+    try {
+      const historyDefinition = getHistoryDefinition(item)
+      return (
+        <PointsHistoryCard {...historyDefinition} testID={`${item.activityId}-${item.createdAt}`} />
+      )
+    } catch (error) {
```

Comment:
yeah definitely. i think it's actually a good idea to do this filtering at the saga level _and_ at the selector level, since there's a possibility that we remove an supported activityId (and its associated rendering code), but records with that activityId still exist within redux

Reply:
wait i'm not quite following...are you saying we want to remove history records if the activityId is no longer supported? this assumption is baked into this PR but i can imagine a world where we stop rewarding points for creating a wallet, but if you've got points for that then we should still show it no? we don't subtract the points from the balance so we should always show them in the history too? this does imply that we'd need to support more activities for the history than the active claimable activities....

Reply:
so, we actually (with this PR) already have two lists of activities; one is for "active activities" (for lack of a better term), or activities that can show up as _cards on the PointsHome_ screen. the other list are these "claim activities", which are _actually_ just activities that can show up as entries on the history screen. we should be able to modify "active activities" in whatever way we want (adding/removing) without affecting the rendering on the history screen.

> i can imagine a world where we stop rewarding points for creating a wallet, but if you've got points for that then we should still show it no

yeah, so we'd remove `create-wallet` from the regular list of activityIds but keep it in the "claim activities" list (which we should definitely rename, maybe like `historicalActivities`)

to address another one of your comments, the reason why it's important to do this check in _both_ the saga and the selector is because of the case where we _remove_ an historical activity that's already in redux state; when we go to render, the handler to render that historical activity will no longer exist, and stuff will break. (this can also be fixed by a fiddly migration if we want to remove an historical activity.) now, i don't think we should really be removing these historical activities to begin with, but i reckon if we can avoid an edge case now, we may as well do it. in fact, it completely suffices to _just_ do this filtering in the selector and remove it from the saga, i think, but it feels weird to knowingly put things into redux that don't match the schema.

Reply:
ooohhhhh i'm sorry, i didn't understand correctly. i was looking at the diff and thought that we only had 1 activityId array, but i see that you have 2 so i think we're aligned.

i think all i wanted to call out is that we could reduce some complexity in the render logic here by ensuring that it doesn't have to do error handling. so the selector's job is to return render-able data, i think it should just return whatever is in redux (implicitly, we want the redux store to not have any data that act as traps for ourselves)

i don't love the idea of having the same filtering logic in the saga _and_ selector because it's a little nuanced to remember why we have that. from what i understand, we can't avoid some kind of filtering logic in the saga because the points config is for all app versions and we need to ignore stuff that we don't understand. you're also saying that we need filtering logic on the selector in case of persisted bad data (e.g. on a new app release, we accidentally stop supporting a previously supported history item type) - is that right? in that case, i think we're better off adding this logic in the REHYDRATE action which is kicked off on app start. i think this would make the use case more obvious than doing it in the selector, wdyt?

Reply:
> you're also saying that we need filtering logic on the selector in case of persisted bad data

yep, that's exactly it - i think sticking this logic in the REHYDRATE action is a much better idea; done!